chainstate/tx-verifier/src/transaction_verifier/tx_index_cache.rs

```diff
@@ -48,10 +48,24 @@ impl TxIndexCache {
         self.data
     }
 
-    pub fn add_tx_index(&mut self, spend_ref: BlockTransactableRef) -> Result<(), TxIndexError> {
+    pub fn add_tx_index(
+        &mut self,
+        spend_ref: BlockTransactableRef,
+        tx_offsets: &[u32],
+    ) -> Result<(), TxIndexError> {
         let tx_index = match spend_ref {
```

Comment:
Please debug_assert here that the lengths are equal.

Reply:
Added, thanks.